src/persist/src/indexed/encoding.rs

```diff
@@ -46,9 +47,16 @@ pub struct BufferEntry<K, V> {
 /// Invariants:
 /// - All strings in id_mapping are unique.
 /// - All ids in id_mapping are unique.
+/// - All strings in graveyard are unique.
+/// - All ids in graveyard are unique.
+/// - None of the strings in graveyard are present in any of the (string, id)
+///   tuples in id_mapping.
+/// - None of the ids in graveyard are present in any of the (string, id) tuples
+///   in id_mapping.
 /// - The same set of ids are present in id_mapping, futures, and traces.
 /// - For each id, the ts_lower in the future is == the ts_upper in the
 ///   corresponding trace.
+/// - |id_mapping| + |graveyard| is equal to next_stream_id.
```

Comment:
oh sorry the |x| syntax was to indicate "cardinality of x" since the previous invariants established that id_mapping and graveyard are sets. I'll change it to .len() to avoid the confusion